src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -474,38 +475,31 @@ def handleJob(self, jobID, fwkJobReport):
             elif jobType == "Express" and set(outputMap).difference(outputModules) == {'write_RAWRAW'}:
                 pass
             else:
-                failJob = True
-                if jobType in ["Processing", "Production"]:
-                    cmsRunSteps = 0
-                    for step in fwkJobReport.listSteps():
-                        if step.startswith("cmsRun"):
-                            cmsRunSteps += 1
-                    if cmsRunSteps > 1:
-                        failJob = False
-
-                if failJob:
+                # Production and Processing filesets do not consider the logArchive step
+                if set(outputMap) != (outputModules - {"logArchive"}):
                     jobSuccess = False
-                    logging.error("Job %d , list of expected outputModules does not match job report, failing job",
-                                  jobID)
-                    logging.debug("Job %d , expected outputModules %s", jobID, sorted(outputMap.keys()))
-                    logging.debug("Job %d , fwjr outputModules %s", jobID, sorted(outputModules))
+                    errMsg = f"Job {jobID}, expected output modules: {sorted(outputMap.keys())}, "
+                    errMsg += f"but has FWJR output modules: {sorted(outputModules)}"
+                    logging.error(errMsg)
+                    # override file list by the logArch1 output only
                     fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
-                else:
-                    logging.warning(
-                        "Job %d , list of expected outputModules does not match job report, accepted for multi-step CMSSW job",
-                        jobID)
-            # Make sure every file has a valid location
-            # see https://github.com/dmwm/WMCore/issues/9353
-            for fwjrFile in fileList:
-                # T0 has analysis file without any location, see:
-                # https://github.com/dmwm/WMCore/issues/9497
-                if not fwjrFile.get("locations") and fwjrFile.get("lfn", "").endswith(".root"):
-                    logging.warning("The following file doesn't have any location: %s", fwjrFile)
-                    jobSuccess = False
-                    break
         else:
             fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
 
+        # Make sure every file has a valid location
+        # see https://github.com/dmwm/WMCore/issues/9353
+        newList = []
+        for fwjrFile in fileList:
+            # T0 has analysis file without any location, see:
+            # https://github.com/dmwm/WMCore/issues/9497
+            if not fwjrFile.get("locations") and fwjrFile.get("lfn", "").endswith(".root"):
+                logging.warning("The following file does not have any location: %s", fwjrFile)
+                jobSuccess = False
```

Comment:
I do not see where jobSuccess is defined outside scope of the for loop, therefore, on line 503 it may fail since it was never setup if it was never hit here.

Reply:
L445 (or first line of this method).